validation/src/main/java/io/micronaut/validation/validator/DefaultValidator.java

```diff
@@ -1330,7 +1330,13 @@ public void keyedValue(String nodeName, Object key, Object keyedValue) {
             Object propertyValue,
             @Nullable DefaultPropertyNode container) {
 
-        final BeanIntrospection<Object> beanIntrospection = getBeanIntrospection(propertyValue);
+        BeanIntrospection<Object> beanIntrospection = null;
+        if (propertyValue != null && !ClassUtils.isJavaLangType(propertyValue.getClass())) {
```

Comment:
I'm not sure if `!ClassUtils.isJavaLangType(propertyValue.getClass()))` is enough. This could break other cases where `@Valid` is being applied. Perhaps a `List<Date>`. You could get around that with `ClassUtils.isJavaBasicType`, however even then I think there is a high potential for this to break applications

Reply:
I agree with @jameskleeh that `isjavaLangType(Class)` doesn't cover everything. Any parameterized type supported by a build-in or custom constraint definition doesn't require introspection. That's easy enough to implement for the simple case of `@Valid List<NotIntrospected> list` but is a bit more complicated for something like `@Valid @FooConstraint(…) List<Foo> list`.